src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -61862,6 +61862,35 @@ static void M()
             );
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72056")]
+        public void Lambda_23()
+        {
+            var comp = CreateCompilation("""
+                using System.Collections.Generic;
+
+                string[]? e = null;
+
+                var arr = new[] {
+                    () => M2(e),
+                    () => M1(e)
+                };
+
+                partial class Program
+                {
+                    static void M1(IEnumerable<string> param) {}
+                }
+                """, options: WithNullableEnable(TestOptions.ReleaseExe));
+
+            comp.VerifyDiagnostics(
+                // (6,11): error CS0103: The name 'M2' does not exist in the current context
+                //     () => M2(e),
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "M2").WithArguments("M2").WithLocation(6, 11),
+                // (7,14): warning CS8604: Possible null reference argument for parameter 'param' in 'void Program.M1(IEnumerable<string> param)'.
+                //     () => M1(e)
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "e").WithArguments("param", "void Program.M1(IEnumerable<string> param)").WithLocation(7, 14)
```

Comment:
Before the change, this warning was not reported, and we hit an assert in `NullableWalker.DebugVerifier` that the `M1(e)` `BoundCall` node was not visited during analysis.